presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -2723,6 +2729,48 @@ public void testGroupedExecution()
             assertQuery(colocatedOneGroupAtATime, chainedSharedBuildOuterJoin, expectedChainedOuterJoinResult, assertRemoteExchangesCount(2));
             assertQuery(colocatedOneGroupAtATimeDynamic, chainedSharedBuildOuterJoin, expectedChainedOuterJoinResult, assertRemoteExchangesCount(2));
 
+            //
+            // Window function
+            // ===============
+            // Window partition key covers bucket key
+            @Language("SQL") String expectedWindowFunctionResult = format("VALUES %s", String.join(", ", nCopies(15000, "(1)")));
+            assertQuery(
+                    colocatedOneGroupAtATime,
+                    "SELECT rank() OVER (PARTITION BY key) FROM test_window",
+                    expectedWindowFunctionResult,
+                    assertRemoteExchangesCount(1));
+
+            assertQuery(
+                    colocatedOneGroupAtATime,
+                    "SELECT row_number() OVER (PARTITION BY key) FROM test_window",
+                    expectedWindowFunctionResult,
+                    assertRemoteExchangesCount(1));
+
+            assertQuery(
+                    colocatedOneGroupAtATime,
+                    "SELECT row_number FROM (SELECT row_number() OVER (PARTITION BY key ORDER BY value) as row_number FROM test_window) WHERE row_number = 1",
```

Comment:
Change `row_number` to `n` in all of these to make the test query shorter. Also capitalize `AS` since it's a keyword (or remove it since it is optional).

Same comment for the similar query below.